### PR TITLE
chore: migrate to Material for MKDocs

### PR DESCRIPTION
**Pull request by hikarukin**
Migrate docs to Material for MKDocs

## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [X] Build-related changes
- [X] Documentation content changes
- [ ] Other (please describe):

Note: build related changes due to adding documentation.yaml workflow that builds and deploys to GH Pages.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by hikarukin** [comment]
#35 should be merged first.

File: docs/development-guide/30_submitting-a-pr.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+---
+
+> First and foremost: as a potential contributor, your changes and ideas are
+> welcome at any hour of the day or night, weekdays, weekends, and holidays.
+> Please do not ever hesitate to ask a question or send a PR.
+
+!!! Tip
+    Before you submit a pull request, please read this document from the Istio
+    documentation which contains very good insights and best practices:
+    ["Writing Good Pull Requests"](https://github.com/istio/istio/wiki/Writing-Good-Pull-Requests).
```
Comment (devotional-phoenix-97) [inline docs/development-guide/30_submitting-a-pr.md]: Maybe we should fork this to our code?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,49 +1,42 @@
...
+dialogue!
 
 ## Ways to contribute
 
 We welcome many types of contributions including:
 
-* New features
-* Builds, CI/CD
-* Bug fixes
-* [Documentation](docs/index.md)
```
Comment (devotional-phoenix-97) [inline CONTRIBUTING.md:29]: Can we replace this with links to our docs site?

File: .github/workflows/documentation.yaml [hunk 3]
```diff
@@ -0,0 +1,88 @@
+name: documentation
+on:
+  push:
+    branches:
+      - main
```
Comment (devotional-phoenix-97) [inline .github/workflows/documentation.yaml:5]: Can we have release based docs?
And should we have release based docs?

File: docs/development-guide/30_submitting-a-pr.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
+---
+title: Submitting a Pull Request
+summary: How to submit a Pull Request
+authors:
+  - devotional-phoenix-97
+  - hikarukin
+date: 2024-07-04
```
Comment (devotional-phoenix-97) [inline docs/development-guide/30_submitting-a-pr.md]: We should update this

File: docs/user-guide/02_adding-extras.md [hunk 5]
```diff
@@ -1,64 +1,12 @@
 ---
-title: Basic usage
+title: Adding extra components
 summary: A description of the basic usage from the perspective of an end user.
```
Comment (devotional-phoenix-97) [inline docs/user-guide/02_adding-extras.md]: Does not match tittle (line 2)

File: docs/user-guide/02_adding-extras.md [hunk 6]
```diff
@@ -1,64 +1,12 @@
...
 authors:
   - hikarukin
 date: 2024-07-04
 ---
 
-Basic Usage
```
Comment (devotional-phoenix-97) [inline docs/user-guide/02_adding-extras.md:9]: Was this duplicate (just checkin')...

**Review by devotional-phoenix-97 — CHANGES_REQUESTED** [review]

File: .github/workflows/documentation.yaml (hunk 3 above)
Comment (devotional-phoenix-97) [reply .github/workflows/documentation.yaml:5]: Fine to do this later.

Comment (hikarukin) [reply .github/workflows/documentation.yaml:5]: Yes, but later.

File: CONTRIBUTING.md (hunk 2 above)
Comment (devotional-phoenix-97) [reply CONTRIBUTING.md:29]: ok to do this later.

File: docs/development-guide/30_submitting-a-pr.md (hunk 4 above)
Comment (hikarukin) [reply docs/development-guide/30_submitting-a-pr.md]: Will do

File: docs/development-guide/30_submitting-a-pr.md (hunk 1 above)
Comment (hikarukin) [reply docs/development-guide/30_submitting-a-pr.md]: I will pull this into out documentation site.

Comment (hikarukin) [reply docs/development-guide/30_submitting-a-pr.md]: done

**Review by devotional-phoenix-97 — DISMISSED** [review]

**Review by devotional-phoenix-97 — APPROVED** [review]

**Pull request merged**